diff --git a/framework/di/Container.php b/framework/di/Container.php
index 5446126..4033f33 100644
--- a/framework/di/Container.php
+++ b/framework/di/Container.php
@@ -428,6 +428,8 @@ class Container extends Component
             foreach ($constructor->getParameters() as $param) {
                 if ($param->isDefaultValueAvailable()) {
                     $dependencies[] = $param->getDefaultValue();
+                } elseif ($param->getName() == "config") {
+                    $dependencies[] = [];
                 } else {
                     $c = $param->getClass();
                     $dependencies[] = Instance::of($c === null ? null : $c->getName());
@@ -542,6 +544,8 @@ class Container extends Component
                     } catch (NotInstantiableException $e) {
                         if ($param->isDefaultValueAvailable()) {
                             $args[] = $param->getDefaultValue();
+                        } elseif ($param->getName() == "config") {
+                            $dependencies[] = [];
                         } else {
                             throw $e;
                         }
@@ -555,6 +559,8 @@ class Container extends Component
                 $args[] = array_shift($params);
             } elseif ($param->isDefaultValueAvailable()) {
                 $args[] = $param->getDefaultValue();
+            } elseif ($name == "config") {
+                $args[] = [];
             } elseif (!$param->isOptional()) {
                 $funcName = $reflection->getName();
                 throw new InvalidConfigException("Missing required parameter \"$name\" when calling \"$funcName\".");
